src/Microsoft.DotNet.Tools.Compiler/Program.cs

```diff
@@ -361,7 +369,7 @@ private static bool CompileProject(ProjectContext context, string configuration,
 
             var success = result.ExitCode == 0;
 
-            if (success && compilationOptions.EmitEntryPoint.GetValueOrDefault())
+            if (success && !noHost && compilationOptions.EmitEntryPoint.GetValueOrDefault())
```

Comment:
This is going to prevent copying the project-to-project dependencies as well. Is that ok? Does `publish` take care of them too when publishing.